merchant.go

```diff
@@ -75,6 +75,11 @@ func MerchantCertificate(cert tls.Certificate) func(*Merchant) error {
 		if _, ok := cert.PrivateKey.(*rsa.PrivateKey); !ok {
 			return errors.New("merchant key should be RSA")
 		}
+
+		if err := checkValidity(cert); err != nil {
```

Comment:
I think the validation should be on the Merchant structure itself after we apply the options but looks like the library follow the validation on the options. (I am ok with this anyway)
Maybe let give a try on staging to point go mod of applepay-app to this branch and see if everything is ok, just in case.

Reply:
I'd say there is no point in attaching invalid certificate, so the validation in option that adds it is good imo ;)

Reply:
yes true, I think the concept about option is wrong because it is not an option it is more required the certificate.
I see the options pattern like without options the library should work with the default values